custom_components/localtuya/cover.py

```diff
@@ -180,108 +201,85 @@ def __init__(
         self,
         device,
         config_entry,
-        switchid,
+        coverid,
         **kwargs,
     ):
-        #_LOGGER.info("running def __init__ of LocaltuyaCover(CoverEntity) with self=%s device=%s name=%s friendly_name=%s icon=%s switchid=%s open_cmd=%s close_cmd=%s stop_cmd=%s", self, device, name, friendly_name, icon, switchid, open_cmd, close_cmd, stop_cmd)
-        super().__init__(device, config_entry, switchid, **kwargs)
-        self._state = None
-        self._position = 50
+        super().__init__(device, config_entry, coverid, **kwargs)
+        self._position = None
+        self._current_cover_position = None
+        self._last_movement = None
+        self._last_position_set = None
+        self._last_command = None
         print(
-            "Initialized tuya cover  [{}] with switch status [{}] and state [{}]".format(
-                self.name, self._status, self._state
+            "Initialized tuya cover  [{}] with switch status [{}]".format(
+                self.name, self._status
             )
         )
 
     @property
-    def available(self):
-        """Return if device is available or not."""
-        return self._available
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        if self._current_cover_position == 100:
+            return True
+        return False
 
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
-        return supported_features
+    def status_updated(self):
+        """Device status was updated."""
+        _LOGGER.debug("status_updated called")
+        self._cached_status = self._status
+        _LOGGER.debug("status_updated called, self._status=%s, self._cached_status=%s", self._status, self._cached_status)
+        try:
```

Comment:
Remember that when you do this (catching exceptions), the cover will still appear as available in the UI and might give the illusion of working, I would argue that it's better that we fail, get an error report and try to solve the problem instead. The logged warning also hides which DP caused the issue.

Reply:
Remember that not all cover devices report the current position, last position set, last command and last movement dp values. I think the best thing right now is to remove the try/except, and then in a future PR, i'll do a specific check to see if the dps exist before trying to update them in status_updated.

Reply:
The you should probably use `.get(...)` instead so that `None` is returned when looking up a missing key. Then we at least try to read all values and save whatever we find.